generator.py

```diff
@@ -77,18 +73,16 @@ def build_gir_if_needed(updated_submodule):
 
 
 def regen_crates(path, conf, level=0):
-    for entry in listdir(path):
-        entry_file = join(path, entry)
-        if isdir(entry_file):
-            if level < 2 and not regen_crates(entry_file, conf, level + 1):
+    for entry in path.iterdir():
+        if entry.is_dir():
+            if level < 2 and not regen_crates(entry, conf, level + 1):
                 return False
-        elif entry.startswith("Gir") and entry.endswith(".toml"):
-            print('==> Regenerating "{}"...'.format(entry_file))
+        elif entry.name.startswith("Gir") and entry.suffix == ".toml":
+            print('==> Regenerating "{}"...'.format(entry))
```

Comment:
Just to confirm: it prints the "full path" right?

Reply:
Yup, `entry.name` is used intentionally in the line above to get just the file name.

Reply:
:+1: